### PR TITLE
feat: add cancel stream function

### PR DESCRIPTION
**Pull request by bertux**
## Description of the changes
Add function to cancel stream paying a request, to be called from Request Finance UI.

### CODE REVIEW

**Comment by bertux** [comment]
I will do the refactoring after EthCC.

File: packages/payment-processor/src/payment/erc777-stream.ts [hunk 1]
```diff
@@ -11,11 +11,12 @@ import {
 import { Framework } from '@superfluid-finance/sdk-core';
 
 export const resolverAddress = '0x913bbCFea2f347a24cfCA441d483E7CBAc8De3Db';
+export const userDataPrefix = '0xbeefac';
 
 /**
  * Processes a transaction to pay an ERC777 stream Request.
  * @param request
- * @param signerOrProvider the Web3 provider, or signer. Defaults to window.ethereum.
+ * @param signer the Web3 provider, or signer. Defaults to window.ethereum.
```
Comment (alexandre-abrioux) [inline packages/payment-processor/src/payment/erc777-stream.ts:19]:
```suggestion
 * @param signer the Web3 signer. Defaults to window.ethereum.
```

File: packages/payment-processor/src/payment/erc777-stream.ts [hunk 2]
```diff
@@ -41,22 +42,90 @@ export async function payErc777StreamRequest(
     signer: signer,
     provider: signer.provider,
   });
-  const superToken = await sf.loadSuperToken(request.currencyInfo.value);
   // FIXME: according to specs PR https://github.com/RequestNetwork/requestNetwork/pull/688
   // in file packages/advanced-logic/specs/payment-network-erc777-stream-0.1.0.md
+  // Below are the SF actions to add in the BatchCall :
```
Comment (alexandre-abrioux) [inline packages/payment-processor/src/payment/erc777-stream.ts:47]:
```suggestion
  // Below are the SF actions to add in the BatchCall:
```

File: packages/payment-processor/src/payment/erc777-stream.ts [hunk 3]
```diff
@@ -41,22 +42,90 @@ export async function payErc777StreamRequest(
     signer: signer,
```
Comment (alexandre-abrioux) [inline packages/payment-processor/src/payment/erc777-stream.ts:42]:
```suggestion
    signer,
```

File: packages/payment-processor/src/payment/erc777-stream.ts [hunk 4]
```diff
@@ -41,22 +42,90 @@ export async function payErc777StreamRequest(
...
-    userData: `0xbeefac${paymentReference}`,
+    userData: `${userDataPrefix}${paymentReference}`,
     overrides: overrides,
   });
+}
+
+/**
+ * Processes a transaction to complete an ERC777 stream paying a Request.
+ * @param request
+ * @param signer the Web3 provider, or signer. Defaults to window.ethereum.
```
Comment (alexandre-abrioux) [inline packages/payment-processor/src/payment/erc777-stream.ts:77]:
```suggestion
 * @param signer the Web3 signer. Defaults to window.ethereum.
```

File: packages/payment-processor/src/payment/erc777-stream.ts [hunk 5]
```diff
@@ -41,22 +42,90 @@ export async function payErc777StreamRequest(
...
+    request.currencyInfo.network === 'private' ? 'custom' : request.currencyInfo.network;
+  const sf = await Framework.create({
+    networkName,
+    provider: signer.provider ?? getProvider(),
+    dataMode: request.currencyInfo.network === 'private' ? 'WEB3_ONLY' : undefined,
+    resolverAddress: request.currencyInfo.network === 'private' ? resolverAddress : undefined,
+    protocolReleaseVersion: request.currencyInfo.network === 'private' ? 'test' : undefined,
+  });
+  const superSigner = sf.createSigner({
+    signer: signer,
```
Comment (alexandre-abrioux) [inline packages/payment-processor/src/payment/erc777-stream.ts:100]:
```suggestion
    signer,
```

File: packages/payment-processor/test/payment/erc777-stream.test.ts [hunk 6]
```diff
@@ -183,18 +187,56 @@ describe('erc777-stream', () => {
...
+        protocolReleaseVersion: 'test',
+      });
+
+      // use the framework to get the SuperToken
+      const daix = await sf.loadSuperToken('fDAIx');
+
+      // wait 2 seconds of streaming to avoid failing
+      await new Promise((r) => setTimeout(r, 2000));
+
+      // Paying fDAIX stream request
```
Comment (alexandre-abrioux) [inline packages/payment-processor/test/payment/erc777-stream.test.ts:222]:
```suggestion
      // Stopping fDAIX stream request
```

**Review by alexandre-abrioux — APPROVED** [review]

**Review by yomarion — APPROVED** [review]
Well done

**Pull request merged**